cpp/src/io/json/write_json.cu

```diff
@@ -0,0 +1,733 @@
+      }
+
+      // write out column's row data (or narep if the row is null)
+      if (include_element) {
+        auto const d_str = is_null_element ? d_narep.value() : d_column.element<string_view>(idx);
+        if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, d_str);
+        bytes += d_str.size_bytes();
+      }
+
+      write_separator = write_separator || include_element;
+    }
+    if (d_buffer) d_buffer = strings::detail::copy_string(d_buffer, row_suffix);
+    bytes += row_suffix.size_bytes();
+
+    if (!d_chars) d_offsets[idx] = bytes;
+  }
+};
+
+/**
+ * @brief Concatenate the strings from each row of the given table as structs in JSON string
+ *
+ * Each row will be struct with field name as column names and values from each column in the table.
+ *
+ * @param strings_columns Table of strings columns
+ * @param column_names Column of names for each column in the table
+ * @param row_prefix Prepend this string to each row
+ * @param row_suffix  Append this string to each row
+ * @param column_name_separator  Separator between column name and value
+ * @param value_separator Separator between values
+ * @param narep Null-String replacement
+ * @param include_nulls Include null string entries in the output
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource to use for device memory allocation.
+ * @return New strings column of JSON structs in each row
+ */
+std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
+                                          column_view const& column_names,
+                                          string_view const row_prefix,
+                                          string_view const row_suffix,
+                                          string_view const column_name_separator,
+                                          string_view const value_separator,
+                                          string_scalar const& narep,
+                                          bool include_nulls,
+                                          rmm::cuda_stream_view stream,
+                                          rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(column_names.type().id() == type_id::STRING, "Column names must be of type string");
+  auto const num_columns = strings_columns.num_columns();
+  CUDF_EXPECTS(num_columns == column_names.size(),
+               "Number of column names should be equal to number of columns in the table");
+  // check all columns are of type string
+  CUDF_EXPECTS(std::all_of(strings_columns.begin(),
+                           strings_columns.end(),
+                           [](auto const& c) { return c.type().id() == type_id::STRING; }),
+               "All columns must be of type string");
+  auto const strings_count = strings_columns.num_rows();
+  if (strings_count == 0)  // empty begets empty
+    return make_empty_column(type_id::STRING);
```

Comment:
Perhaps move this check up. No need to check the types if there are no rows.
Also, what happens if there are no columns?

Reply:
return empty string column.